File: dotnet/cafdemalihapa/Directories/Repository/Dotnet/Cafdemalihapa/Scripts/EnvironmentVariablesSource/CodeBases/MutantVariablesFile.cs
using cafdemalihapa.Applications.Cafdemalihapa;
using cafdemalihapa.Directories.Repository.Dotnet.Cafdemalihapa.Scripts.EnvironmentVariablesSource.CodeBases;
using cafdemalihapa.Names;

namespace cafdemalihapa.Directories.Repository.Dotnet.Cafdemalihapa.Scripts.EnvironmentVariablesSource.Files.Jsons
{
    public class MutantVariablesFile(
            PrimaryApplication primaryApplication,
            SecondaryApplication secondaryApplication,
            CodeBaseDirectory codeBaseDirectory
        )
    {
        public string GetName()
        {
            var primaryApplicationName = primaryApplication.GetName();
            var secondaryApplicationName = secondaryApplication.GetName();
            var fileName = $"{primaryApplicationName}-{secondaryApplicationName}.json";
            return fileName;
        }

        public string GetPath()
        {
            var name =GetName();
            return Path.Combine(
                codeBaseDirectory.GetPath(), name);
        }
    }
}

File: dotnet/cafdemvers/zzq-cafdemwimu/cafdemwimu.console/Directories/Repository/Dotnet/Cafdemalihapa/Scripts/EnvironmentVariablesSource/CodeBases/MutantVariablesFile.fs
namespace cafdemwimu.console.Directories.Repository.Dotnet.Cafdemalihapa.Scripts.EnvironmentVariablesSource.Files.Jsons

open System.IO
open cafdemwimu.console.Names
open cafdemwimu.console.Directories.Repository.Dotnet.Cafdemalihapa.Scripts.EnvironmentVariablesSource.CodeBases

type MutantVariablesFile(primaryApplication: PrimaryApplication, secondaryApplication: SecondaryApplication, codeBaseDirectory: CodeBaseDirectory) =
    member _.GetName() =
        let primaryApplicationName = primaryApplication.GetName()
        let secondaryApplicationName = secondaryApplication.GetName()
        let fileName = $"{primaryApplicationName}-{secondaryApplicationName}.json"
        fileName

    member this.GetPath() =
        let name = this.GetName()
        Path.Combine(codeBaseDirectory.GetPath(), name)
